R/list_comparison_linter.R

```diff
@@ -0,0 +1,69 @@
+#' Block usage of comparison operators with known-list() functions like lapply
+#'
+#' Usage like `lapply(x, sum) > 10` is awkward because the list must first
+#'   be coerced to a vector for comparison. A function like [vapply()]
+#'   should be preferred.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "lapply(x, sum) > 10",
+#'   linters = list_comparison_linter()
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "unlist(lapply(x, sum)) > 10",
+#'   linters = list_comparison_linter()
+#' )
+#'
+#' @evalRd rd_tags("list_comparison_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+list_comparison_linter <- function() {
+  # TODO(michaelchirico): extend to cases where using simplify=FALSE implies a
+  #   list output, e.g. with sapply, replicate, mapply.
```

Comment:
@AshesITR  it's under the TODO -- `mapply()` may or may not return a list. e.g. ``mapply(`+`, 1:10, 1:10)``.

Reply:
Does `.mapply()` behave differently?

Reply:
`.mapply()` always returns list, see `mapply()` implementation which basically calls `simlify2array()` on `.mapply()` output:

https://github.com/r-devel/r-svn/blob/9f3b4bea25f93be3ae951e40f36bcf8e54a7201f/src/library/base/R/mapply.R#L19-L38

I'll add it